src/status_im/ui/screens/chat/components/input.cljs

```diff
@@ -73,27 +77,102 @@
-  [rn/view {:style (styles/text-input-wrapper)}
-   [rn/text-input {:style                  (styles/text-input)
-                   :ref                    text-input-ref
-                   :maxFontSizeMultiplier  1
-                   :accessibility-label    :chat-message-input
-                   :text-align-vertical    :center
-                   :multiline              true
-                   :default-value          text-value
-                   :editable               (not cooldown-enabled?)
-                   :blur-on-submit         false
-                   :auto-focus             false
-                   :on-focus               #(set-active-panel nil)
-                   :on-change              #(on-text-change (.-text ^js (.-nativeEvent ^js %)))
-                   :max-length             chat.constants/max-text-size
-                   :placeholder-text-color (:text-02 @colors/theme)
-                   :placeholder            (if cooldown-enabled?
-                                             (i18n/label :cooldown/text-input-disabled)
-                                             (i18n/label :t/type-a-message))
-                   :underlineColorAndroid  :transparent
-                   :auto-capitalize        :sentences}]])
+(defn text-input
+  [{:keys [cooldown-enabled? text-value on-text-change set-active-panel text-input-ref]}]
+  (let [cursor            @(re-frame/subscribe [:chat/cursor])
+        mentionable-users @(re-frame/subscribe [:chats/mentionable-users])]
+    [rn/view {:style (styles/text-input-wrapper)}
+     [rn/text-input
+      {:style                  (styles/text-input)
+       :ref                    text-input-ref
+       :maxFontSizeMultiplier  1
+       :accessibility-label    :chat-message-input
+       :text-align-vertical    :center
+       :multiline              true
+       :default-value          text-value
+       :editable               (not cooldown-enabled?)
+       :blur-on-submit         false
+       :auto-focus             false
+       :on-focus               #(set-active-panel nil)
+       :max-length             chat.constants/max-text-size
+       :placeholder-text-color (:text-02 @colors/theme)
+       :placeholder            (if cooldown-enabled?
+                                 (i18n/label :cooldown/text-input-disabled)
+                                 (i18n/label :t/type-a-message))
+       :underlineColorAndroid  :transparent
+       :auto-capitalize        :sentences
+       :selection
+       (when (and cursor platform/android?)
+         (clj->js
+          {:start cursor
+           :end   cursor}))
+
+       :on-selection-change
+       (fn [_]
+         (when cursor
+           (re-frame/dispatch [::mentions/clear-cursor])))
+
+       :on-change
+       (fn [args]
+         (let [text (.-text ^js (.-nativeEvent ^js args))]
+           (on-text-change text)
+           (when platform/ios?
```

Comment:
could you pls elaborate why this is only for ios ?

Reply:
Yep, I will add a comment there, the problem is that event object contains different data on Android and iOS.

Reply:
oops wrong reason, this specific condition is because of order of events. But I will add comments in all places where we have platform based conditions in this PR

Reply:
added notes